src/Symfony/Component/Translation/Dumper/YamlFileDumper.php

```diff
@@ -36,4 +84,73 @@ protected function getExtension()
     {
         return 'yml';
     }
+
+    private function expandToTree(array $messages)
+    {
+        $tree = array();
+
+        foreach ($messages as $id => $value) {
+            $this->expandElement($tree, explode('.', $id), $value);
```

Comment:
And tree should work correctly in this case because storing translation as tree just improve readability of file. It will be restored to linear structure while loading back in message catalogue.